Implement AEIP2 update

Neylix opened the pull request:
# Description

Implementation of the new version of AEIP2 (https://github.com/archethic-foundation/aeip/commit/71d6e4f76e22a426b1b4ddf1f086dfe31359ffa4)

Fixes https://github.com/archethic-foundation/aeip/issues/4 / https://github.com/archethic-foundation/aeip/issues/9

## Type of change

- New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Update test files
Creation of multiple NFT with or without collection, id, properties ...

# Checklist:

- My code follows the style guidelines of this project
- I have performed a self-review of my own code
- I have commented my code, particularly in hard-to-understand areas
- I have made corresponding changes to the documentation
- My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- New and existing unit tests pass locally with my changes
- Any dependent changes have been merged and published in downstream modules

samuelmanzanera commented on lib/archethic/mining/fee.ex:
```diff
@@ -72,20 +72,16 @@ defmodule Archethic.Mining.Fee do
     with {:ok, json} <- Jason.decode(content),
          "non-fungible" <- Map.get(json, "type", "fungible"),
-         utxos = [_ | _] <- Map.get(json, "properties", []) do
+         utxos when is_list(utxos) <- Map.get(json, "collection") do
```
The former change was to ensure the list is not empty without iterating over the entire to know the size, maybe we should keep it to prevent empty NFT collection.

samuelmanzanera replied:
ok

samuelmanzanera reviewed (commented):
We might also change a bit the explorer to display—based on the decimals—the number of tokens received.

Neylix commented:
> We might also change a bit the explorer to display—based on the decimals—the number of tokens received.

Updated explorer UI to handle decimals, and display token symbol instead of address

Also fixed bug of #560

samuelmanzanera commented on lib/archethic_web/live/transaction_details_live.ex:
```diff
@@ -78,32 +83,86 @@ defmodule ArchethicWeb.TransactionDetailsLive do
+    uco_price_at_time = tx.validation_stamp.timestamp |> OracleChain.get_uco_price()
+    uco_price_now = DateTime.utc_now() |> OracleChain.get_uco_price()
+
+    tokens =
```
If you want to reduce lines and avoid repetition you could use recursive function:

```elixir

token_properties = []
 |> get_token_addresses(ledger_inputs)
 |> get_token_addresses(transaction_movements)
 |> get_token_addresses(token_transfers)
 |> Enum.uniq()
 |> get_token_properties()

# ...

defp get_token_addresses(acc, [ %TransactionMovement{ type: {:token, token_address, _}} | rest. ]) do
    get_token_addresses([ token_address | acc ], rest)
end

defp get_token_addresses(acc, [ %TransactionInput{ type: {:token, token_address, _}} | rest. ]) do
    get_token_addresses([ token_address | acc ], rest)
end

defp get_token_addresses([ %TokenTransfer{ token_address: token_address } | rest. ], acc) do
    get_token_addresses([ token_address | acc ], rest)
end

defp get_token_addresses(acc, _) do
    get_token_addresses(acc, rest)
end


```
